### PR TITLE
Avoid message body together with 204, 304 and 1xx status codes

### PR DESCRIPTION
**Pull request by anisimkov**
GitHub: AWS.Client.Post hangs on S204 No Content #23

### CODE REVIEW

File: src/core/aws-server-http_utils.adb [hunk 1]
```diff
@@ -201,10 +201,8 @@ package body AWS.Server.HTTP_Utils is
                --  If-Match/If-None-Match and ETag values.
 
                if Is_Ignored (Answer) then
-                  Answer := Response.Build
-                    (Status_Code   => Messages.S304,
-                     Content_Type  => "text/plain",
-                     Message_Body  => "Value is not modified (ETag)");
+                  Answer := Response.Acknowledge
+                              (Status_Code => Messages.S304);
```
Comment (TurboGit) [inline src/core/aws-server-http_utils.adb]: can probably fit on the line above

File: src/core/aws-messages.ads [hunk 2]
```diff
@@ -204,6 +204,10 @@ package AWS.Messages is
    function Reason_Phrase (S : Status_Code) return String;
    --  Returns the reason phrase for the status code S, see [RFC 2616 - 6.1.1]
 
+   function With_Body (S : Status_Code) return Boolean is
```
Comment (TurboGit) [inline src/core/aws-messages.ads]: expression function definition should be in the private part. only the spec visible.

File: src/core/aws-client-http_utils.adb [hunk 3]
```diff
@@ -336,12 +336,23 @@ package body AWS.Client.HTTP_Utils is
       Parse_Header (Connection, Result, Keep_Alive);
 
       declare
-         TE     : constant String  :=
```
Comment (TurboGit) [inline src/core/aws-client-http_utils.adb]: I would avoid changing that, this is just add use clause which does not make the code clearer to me.

**Comment by TurboGit** [comment]
Apart from minor style comments, the patch looks good to me.

File: src/core/aws-server-http_utils.adb (hunk 1 above)
Comment (anisimkov) [reply src/core/aws-server-http_utils.adb]: Moved

File: src/core/aws-messages.ads (hunk 2 above)
Comment (anisimkov) [reply src/core/aws-messages.ads]: Done

File: src/core/aws-client-http_utils.adb (hunk 3 above)
Comment (anisimkov) [reply src/core/aws-client-http_utils.adb]: use removed, package prefixes restored

**Comment by anisimkov** [comment]
3 comments satisfied I hope.

File: src/core/aws-messages.ads [hunk 4]
```diff
@@ -204,6 +204,9 @@ package AWS.Messages is
    function Reason_Phrase (S : Status_Code) return String;
    --  Returns the reason phrase for the status code S, see [RFC 2616 - 6.1.1]
 
+   function With_Body (S : Status_Code) return Boolean;
+   --  Returns True if message with status can has the body
```
Comment (TurboGit) [inline src/core/aws-messages.ads]: --  ... can have a body...

File: src/core/aws-server-http_utils.adb [hunk 5]
```diff
@@ -1310,6 +1307,7 @@ package body AWS.Server.HTTP_Utils is
                        Line_Attribute.Reference;
 
       Status_Code : Messages.Status_Code := Response.Status_Code (Answer);
+      With_Body   : constant Boolean := Messages.With_Body (Status_Code);
```
Comment (TurboGit) [inline src/core/aws-server-http_utils.adb]: I always put the constant before the variable.

Comment (anisimkov) [reply src/core/aws-server-http_utils.adb]: This constant here because uses just declared variable.

File: src/core/aws-messages.ads (hunk 4 above)
Comment (anisimkov) [reply src/core/aws-messages.ads]: Done

File: src/core/aws-server-http_utils.adb (hunk 5 above)
Comment (anisimkov) [reply src/core/aws-server-http_utils.adb]: I'll move it to Send_Data and style will be fixed

Comment (anisimkov) [reply src/core/aws-server-http_utils.adb]: Done

**Comment by anisimkov** [comment]
By the way a bit optimization in change Status_Code to 304, 404 and 206

**Pull request merged**